src/util_exec.cpp

```diff
@@ -522,86 +522,6 @@ Util::freePatchedArgv(void *ptr)
   JALLOC_HELPER_FREE(ptr);
 }
 
-void
-Util::runMtcpRestore(int is32bitElf,
-                     const char *path,
-                     int fd,
-                     size_t argvSize,
-                     size_t envSize)
-{
-  static string mtcprestart = Util::getPath("mtcp_restart");
-
-  if (is32bitElf) {
-    mtcprestart = Util::getPath("mtcp_restart-32", is32bitElf);
-  }
-
-  // Tell mtcp_restart process to write its debugging information to
-  // PROTECTED_STDERR_FD. This way we prevent it from spitting out garbage onto
-  // FD_STDERR if it is being used by the user process in a special way.
-  char protected_stderr_fd_str[16];
-  sprintf(protected_stderr_fd_str, "%d", PROTECTED_STDERR_FD);
-
-  char buf[64];
-  sprintf(buf, "%d", fd);
-
-  char *newArgs[] = {
-    (char *)mtcprestart.c_str(),
-    (char *)"--stderr-fd",
-    protected_stderr_fd_str,
-    (char *)"--fd",
-    buf,
-    NULL
-  };
-  JTRACE("launching mtcp_restart --fd")(fd)(path);
-
```

Comment:
This FIXME could be needed in certain cases.  Removing it completely means mtcp_restart can break in some cases, and we won't have this FIXME on how to fix it.

See:  `mtcp/mtcp_check_vdso.ic` and look at `return setPersonalityEnv("MTCP_OLDPERS", oldpers_str, environ);`.

The issue is that we don't want to use libc in mtcp_restart.c.  So, we can't do `setenv()` there.  But we need the environment variable `"MTCP_OLDPERS"` in case we need to record the old personality and then exec into a new personality for things like removing the address space randomization.  So, we want to create the environment variable `"MTCP_OLDPERS"` in advance, inside `dmtcp_restart.cpp` using `setenv()`, so that the environment variable will later be available to `mtcp_restart`.

There's a comment  here in the `FIXME` about adding an explanation of the logic.  You might want to copy some of the above paragraph into a comment explaining the logic.

Reply:
Hi @gc00, this code was stale and not called from anywhere. This was from an earlier model when we used to create child processes _after_ restoring the parent process. We switched to the current model of forking child processes _before_ running mtcp_restart 7-8 years ago but forgot to remove this code from util_exec.cpp. The personality handling code is currently in all inside src/mtcp/. If you like, we can add a new comment somewhere about personality restoration logic.

Reply:
Here is the original commit for reference: https://github.com/dmtcp/dmtcp/commit/9bf02cf695e31077309c94f3f9235100d97fda0b

Reply:
I looked at `setPersonalityEnv()` definition, and if I am reading it correctly, it _does_ have and equivalent of setenv in the code. Here is the relevant part: https://github.com/dmtcp/dmtcp/blob/master/src/mtcp/mtcp_check_vdso.ic#L386-L403

The current call sequence is: mtcp_check_vdso() -> setenv_oldpers() -> setPersonalityEnv(), so I don't think it's obsolete yet :).